uaa/src/test/java/org/cloudfoundry/identity/uaa/orchestrator/UaaOrchestratorClaimTest.java

```diff
@@ -0,0 +1,301 @@
+                getZoneByName(zoneName, HttpStatus.NOT_FOUND);
+            }
+        }
+    }
+
+
+    // end region Positive Tests
+
+    // end region Negative Tests
+
+    private boolean createZone(final String claimName, final UaaRequest uaaRequest)
+            throws ServiceInstanceProviderException, ApiException {
+        try {
+            orchestratorUtil.createService(claimName, ServiceName.UAA, OBJECT_MAPPER.valueToTree(uaaRequest), ApiEnvironment.valueOf(environment));
+            return true;
+        } catch (ServiceInstanceProviderException | ApiException e) {
+            LOGGER.error("UAA Claim create failed for claimName: {}, uaaRequest:{}, errorMessage: {}",
+                    claimName, uaaRequest, e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    private Claim getZone(final String claimName, final ServiceName serviceName) throws ServiceInstanceProviderException {
+        try {
+            return orchestratorUtil.readClaim(claimName, ServiceName.UAA, ApiEnvironment.valueOf(environment));
+        } catch (ServiceInstanceProviderException e) {
+            LOGGER.error("UAA Claim read failed for claimName: {}, errorMessage: {}", claimName,
+                    e.getMessage(), e);
+            throw e;
+        }
+    }
+
+
+    private boolean deleteZone(final String claimName, final ServiceName serviceName) throws ServiceInstanceProviderException {
+        try {
+            orchestratorUtil.deleteClaim(claimName, ServiceName.UAA, ApiEnvironment.valueOf(environment));
+            return true;
+        } catch (ServiceInstanceProviderException e) {
+            LOGGER.error("UAA Claim delete failed for claimName: {}, errorMessage: {}", claimName,
+                    e.getMessage(), e);
+            throw e;
+        }
+    }
+
+    private String getZoneByName(final String zoneName, HttpStatus status) throws Exception {
+        if (StringUtils.isEmpty(zoneName)) {
+            throw new Exception("zoneName can not be null");
+        }
+        String url = uaaUrl + ZONES_URI + "?name=" + zoneName;
+
+        return executeRequest(getTrustedIssuerValidScopesRestTemplate(), url, HttpMethod.GET, headers, status, null,
+                null);
+    }
+
+    public static String executeRequest(OAuth2RestOperations tenantRestTemplate, String url, HttpMethod method,
+                                        HttpHeaders headers, HttpStatus expectedStatusCode,
+                                        String expectedMessage, String requestBody) {
+        String body = null;
+        if (!HttpMethod.GET.equals(method)) {
+            body = requestBody;
+        }
+        HttpEntity<String> request = new HttpEntity<>(body, headers);
+        log.trace("Executing HTTP {} - url: {}, request:\n {}", method, url, request);
+        if (!expectedStatusCode.is2xxSuccessful()) {
+            ResponseEntity<RestErrorResponse> responseEntity = tenantRestTemplate.exchange(url, method, request,
```

Comment:
Here parsing will fail to RestErrorResponse class because response json format is different in uaa orchestrator zone apis. Refer uaa orchestrator zone apis contract, https://devcloud.swcoe.ge.com/devspace/display/RMOCM/Orchestrator+UAA+Zone+provisioning+API#OrchestratorUAAZoneprovisioningAPI-ZoneAPI

Reply:
Done